internal/delete/delete_item_action_handler.go

```diff
@@ -79,7 +80,16 @@ func InvokeDeleteActions(ctx *Context) error {
 	for resource := range backupResources {
 		gvr, _, err := ctx.DiscoveryHelper.ResourceFor(schema.ParseGroupResource(resource).WithVersion(""))
 		if err != nil {
-			return errors.Wrapf(err, "failed to resolve resource into complete group/version/resource: %v", resource)
+			if !meta.IsNoMatchError(err) {
+				return errors.Wrapf(err, "failed to resolve resource into complete group/version/resource: %v", resource)
+			}
+			// The resource may not be fully resolved via discovery because it is possible that the
+			// resource is a custom resource while the custom resource definition is already removed.
+			// Skip the resolving from discovery here for this case.
+			// Refer to https://github.com/vmware-tanzu/velero/issues/4009 for more details
+			gvr = schema.ParseGroupResource(resource).WithVersion("")
+
+			ctx.Log.Infof("got \"no matches\" error when trying to resolve the resource %s via discovery, use ParseGroupResource directly, gvr: %s", resource, gvr.String())
```

Comment:
So, we can't use the gvr variable on line 81 directly, right?

Reply:
Do you mean to replace `gvr, _, err := ctx.DiscoveryHelper.ResourceFor(schema.ParseGroupResource(resource).WithVersion(""))` with `gvr := schema.ParseGroupResource(resource).WithVersion("")` in line 81? 
I'm not sure whether the input `resource` is fully resolved or not. Handle this when getting errors may be more safe?

Reply:
I have the same question, @ywk253100 could you please check with Nolan.
Seems the resources in the backup tarball should be resolved already?